src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/FilterLongReadAlignmentsSAMSpark.java

```diff
@@ -61,6 +62,12 @@
     private static final long serialVersionUID = 1L;
     private final Logger localLogger = LogManager.getLogger(FilterLongReadAlignmentsSAMSpark.class);
 
+    @Advanced
+    @Argument(doc = "parameter (float) for specifying of two configurations differ by no more than this amount, the two are considered equally good",
+            shortName = "cst",
+            fullName = "configScoreDiffTol", optional = true)
+    public final Double configScoreDiffTolerance = 0.0;
```

Comment:
Do you want the default to be 0.0? I thought you'd want some small delta.

Reply:
After reading the link I gave there, it seems there's no universally good answer to comparing floating point numbers.
What I am trying to fix here is that sometimes (we have no way of knowing when this happens as exact machine representation of floating point numbers depends on a lot of factors) two configurations that should be equally good while calculated by human with paper/pencil could have perfectly different machine representations. 
While a user can provide an epsilon, the code actually check and use which one is bigger&mdash;the user provided epsilon, and one ULP, in case a user provided literal 0 (that is really not desirable). So even though 0 is the default value here, it is actually not used. 
And this feature is marked as `@Advanced` so shouldn't be messed with unless the caller is experimenting with the filtering module itself.




To quote (I think we are facing the 2nd scenario)


>There is no silver bullet. You have to choose wisely.
>
>* If you are comparing against zero, then relative epsilons and ULPs based comparisons are usually meaningless. You’ll need to use an absolute epsilon, whose value might be some small multiple of FLT_EPSILON and the inputs to your calculation. Maybe.
>* If you are comparing against a non-zero number then relative epsilons or ULPs based comparisons are probably what you want. You’ll probably want some small multiple of FLT_EPSILON for your relative epsilon, or some small number of ULPs. An absolute epsilon could be used if you knew exactly what number you were comparing against.
>* If you are comparing two arbitrary numbers that could be zero or non-zero then you need the kitchen sink. Good luck and God speed.

Maybe Kronecker was right: _"God made the integers, all else is the work of man."_. :confused:

Reply:
Sure, but I was just wondering if there was some "effective" difference that's larger than the ULP-type differences but at which point it doesn't matter if the difference is due to floating point representations or the actual input values. For example, if there are two configurations made of five alignments and the only difference in score is due to one containing an alignment with mapq 51 and the other replaces it with an alignment with mapq 52, we might actually want to treat those as roughly equivalent and run them through your disambiguation procedure, no? So whatever that delta is might be an acceptable epsilon for this method. Anyway, this is not a big deal, so feel free to do whatever you want.